overlord/hookstate/ctlcmd/get.go

```diff
@@ -120,3 +135,126 @@ func (c *getCommand) Execute(args []string) error {
 
 	return nil
 }
+
+func (c *getCommand) Execute(args []string) error {
+	context := c.context()
+	if context == nil {
+		return fmt.Errorf("cannot get without a context")
+	}
+
+	if c.Typed && c.Document {
+		return fmt.Errorf("cannot use -d and -t together")
+	}
+
+	var snapAndPlugOrSlot snap.SnapAndName
+	// treat PlugOrSlotSpec argument as config key if it doesn't contain ':'
+	if !strings.Contains(c.Positional.PlugOrSlotSpec, ":") {
+		c.Positional.Keys = append([]string{c.Positional.PlugOrSlotSpec}, c.Positional.Keys[0:]...)
+		c.Positional.PlugOrSlotSpec = ""
+	} else {
+		snapAndPlugOrSlot.UnmarshalFlag(c.Positional.PlugOrSlotSpec)
+	}
+
+	if snapAndPlugOrSlot.Name != "" {
+		return c.handleGetInterfaceAttributes(context, snapAndPlugOrSlot.Snap, snapAndPlugOrSlot.Name)
+	}
+
+	if c.ForcePlugSide || c.ForceSlotSide {
+		return fmt.Errorf("cannot use --plug or --slot without <snap>:<plug|slot> argument")
+	}
+
+	context.Lock()
```

Comment:
Can we please move the rest below to `c.getConfigSetting`, so Execute becomes the preamble to both.